### PR TITLE
Fixes for typescript plugin

### PR DESCRIPTION
**Pull request by tolusha**
### What does this PR do
I introduced couple of fixes to deploy VS Code typescript extension [1] as a plugin 

### Reference issue
https://github.com/theia-ide/theia/issues/3767

[1] https://github.com/Microsoft/vscode/tree/master/extensions/typescript-language-features

### CODE REVIEW

**Comment by tolusha** [comment]
This PR introduces some experimental code from [1]
Related issue https://github.com/theia-ide/theia/issues/4167

[1] https://github.com/Microsoft/vscode/blob/master/src/vs/vscode.proposed.d.ts#L1033-L1035

File: packages/plugin/src/theia.d.ts [hunk 1]
```diff
@@ -1649,6 +1655,21 @@ declare module '@theia/plugin' {
         lineNumbers?: TextEditorLineNumbersStyle;
     }
 
+    // Experimental API
+    // https://github.com/Microsoft/vscode/blob/1.30.2/src/vs/vscode.proposed.d.ts#L1015
+    export interface FileRenameEvent {
```
Comment (akosyakov) [inline packages/plugin/src/theia.d.ts]: @tolusha @benoitf How about keeping vscode.proposed feature into theia.proposed that we don't need to deprecate anything later unnecessary?

**Comment by benoitf** [comment]
@akosyakov yes this is what I said also but forgot to push my review

**Comment by tolusha** [comment]
@akosyakov @benoitf 
It is ready to review again

**Pull request merged**